### PR TITLE
Do not advertise workload updates on run output

### LINKED ISSUE #22571: Suppress workload update notification from `dotnet run`
If workload updates are available, we print the message `Workload updates are available. Run `dotnet workload list` for more information.`

We should probably avoid doing this for the `dotnet run` command, as we typically try to avoid outputting anything that doesn't come from the project being run.

### PR DESCRIPTION
**Pull request by sfoslund**
Fixes https://github.com/dotnet/sdk/issues/22571

### CODE REVIEW

**Comment by sfoslund** [comment]
@baronfel tagging you in case you have any other thoughts on when we should output this message. Right now it's outputted any time one of your installed workloads has an update  on any restoring command (with this change, excluding run). @marcpopMSFT suggested possibly only writing the message when the current project uses the workload, but I'm not sure that would be very easy to implement and I think it would have significant perf impacts.

**Comment by baronfel** [comment]
That seems fine to me, @sfoslund. IF we had https://github.com/dotnet/msbuild/issues/3911 or similar we might be able to add information about the workloads to the invocation of the `SeparateRestoreCommand` as sort of an all-in-one approach, but in lieu of that this is a good change.

File: src/Tests/dotnet-workload-install.Tests/GivenWorkloadManifestUpdater.cs [hunk 1]
```diff
@@ -305,9 +305,10 @@ public void GivenWorkloadManifestUpdateItChoosesHighestManifestVersionInCache()
         }
 
         [Theory]
-        [InlineData("build")]
-        [InlineData("publish")]
-        public void GivenWorkloadsAreOutOfDateUpdatesAreAdvertisedOnRestoringCommands(string commandName)
+        [InlineData("build", false)]
+        [InlineData("publish", false)]
+        [InlineData("run", true)]
+        public void GivenWorkloadsAreOutOfDateUpdatesAreAdvertisedOnRestoringCommands(string commandName, bool commandOptedOut)
```
Comment (dsplaisted) [inline src/Tests/dotnet-workload-install.Tests/GivenWorkloadManifestUpdater.cs:311]: `commandOptedOut` ends up being sort of a double negative.  I would invert the boolean values and call the variable something like `shouldShowUpdateAvailable`.

File: src/Tests/dotnet-workload-install.Tests/GivenWorkloadManifestUpdater.cs [hunk 2]
```diff
@@ -320,14 +321,27 @@ public void GivenWorkloadsAreOutOfDateUpdatesAreAdvertisedOnRestoringCommands(st
...
+                    .HaveStdOutContaining(Workloads.Workload.Install.LocalizableStrings.WorkloadUpdatesAvailable);
+            }
+            else
+            {
+                commandResult
+                    .Should()
+                    .Pass()
+                    .And
+                    .NotHaveStdOutContaining(Workloads.Workload.Install.LocalizableStrings.WorkloadUpdatesAvailable);
+            }
```
Comment (dsplaisted) [inline src/Tests/dotnet-workload-install.Tests/GivenWorkloadManifestUpdater.cs:343]: I think you can do the following if you want to deduplicate the code that checks that the command passed.

```suggestion
            commandResult
                .Should()
                .Pass()

            if (!commandOptedOut)
            {
                commandResult
                    .Should()
                    .HaveStdOutContaining(Workloads.Workload.Install.LocalizableStrings.WorkloadUpdatesAvailable);
            }
            else
            {
                commandResult
                    .Should()
                    .NotHaveStdOutContaining(Workloads.Workload.Install.LocalizableStrings.WorkloadUpdatesAvailable);
            }
```

**Review by dsplaisted — APPROVED** [review]
Looks good, though it looks like there may be some test failures.

**Comment by sfoslund** [comment]
> Looks good, though it looks like there may be some test failures.

This PR needs eab7a38 too, CI should pass once that gets merged along with https://github.com/dotnet/sdk/pull/22557

**Pull request merged**